synthtool/gcp/common.py

```diff
@@ -59,6 +59,12 @@ def _generic_library(self, directory: str, **kwargs) -> Path:
 
         return result
 
+    def py_samples(self, **kwargs) -> Path:
+        # kwargs["metadata"] is required to load values from .repo-metadata.json
+        if "metadata" not in kwargs:
+            kwargs["metadata"] = {}
```

Comment:
Why is it necessary to set kwargs["metadata"] like this?

Reply:
I'm not completely sure, I leveraged existing code from the rest of common.py. Perhaps [_generic_library()](https://github.com/googleapis/synthtool/blob/b962b480c5cf5146752ac50d0084fc53dec734af/synthtool/gcp/common.py#L47) isn't built to handle nonexistent metadata as is.

Reply:
How does the code behave if you remove lines 63-65?

Reply:
![test_kwargs](https://user-images.githubusercontent.com/39810066/87955620-fea16680-ca7b-11ea-8265-8f5dd470d623.png)
I wrote a simple test to demonstrate what happens if there is null metadata. From what I can tell it seems like Jinja itself does not have the ability to handle undefined metadata.

Reply:
![test_kwargs_code](https://user-images.githubusercontent.com/39810066/87956041-88e9ca80-ca7c-11ea-820e-1a0ebde8c985.png)